src/vellum/workflows/nodes/core/map_node/node.py

```diff
@@ -75,95 +75,85 @@ def run(self) -> Iterator[BaseOutput]:
             return
 
         self._event_queue: Queue[Tuple[int, WorkflowEvent]] = Queue()
-        self._concurrency_queue: Queue[Thread] = Queue()
-        fulfilled_iterations: List[bool] = []
-        for index, item in enumerate(self.items):
-            fulfilled_iterations.append(False)
-            current_execution_context = get_execution_context()
-            thread = Thread(
-                target=self._context_run_subworkflow,
-                kwargs={
-                    "item": item,
-                    "index": index,
-                    "current_execution_context": current_execution_context,
-                },
-            )
-            if self.max_concurrency is None:
-                thread.start()
-            else:
-                self._concurrency_queue.put(thread)
-
-        if self.max_concurrency is not None:
-            concurrency_count = 0
-            while concurrency_count < self.max_concurrency:
-                is_empty = self._start_thread()
-                if is_empty:
-                    break
-
-                concurrency_count += 1
-
-        try:
-            while map_node_event := self._event_queue.get():
-                index = map_node_event[0]
-                subworkflow_event = map_node_event[1]
-                self._context._emit_subworkflow_event(subworkflow_event)
-
-                if not is_workflow_event(subworkflow_event):
-                    continue
-
-                if subworkflow_event.workflow_definition != self.subworkflow:
-                    continue
-
-                if subworkflow_event.name == "workflow.execution.initiated":
-                    for output_name in mapped_items.keys():
-                        yield BaseOutput(name=output_name, delta=(None, index, "INITIATED"))
-
-                elif subworkflow_event.name == "workflow.execution.fulfilled":
-                    for output_reference, output_value in subworkflow_event.outputs:
-                        if not isinstance(output_reference, OutputReference):
-                            logger.error(
-                                "Invalid key to map node's subworkflow event outputs",
-                                extra={"output_reference_type": type(output_reference)},
+        fulfilled_iterations: List[bool] = [False] * len(self.items)
+
+        max_workers = self.max_concurrency if self.max_concurrency is not None else len(self.items)
+
+        with concurrent.futures.ThreadPoolExecutor(max_workers=max_workers) as executor:
+            futures = []
+            for index, item in enumerate(self.items):
+                current_execution_context = get_execution_context()
```

Comment:
[nit] this should actually be the same for all iterations and could be lifted out of the loop